In [1]:
import warnings
warnings.filterwarnings("ignore")
import intake

In [2]:
#col = intake.open_esm_metadatastore(collection_input_definition="CESM1-LE", overwrite_existing=True)
col = intake.open_esm_metadatastore(collection_name="CESM1-LE")

In [3]:
col.df.head()

resource resource_type  \
0  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
1  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
2  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
3  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
4  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   

   direct_access experiment                         case component  stream  \
0           True       CTRL  f.e11.F1850C5CN.f09_f09.001       atm  cam.h1   
1           True       CTRL  f.e11.F1850C5CN.f09_f09.001       atm  cam.h1   
2           True       CTRL  f.e11.F1850C5CN.f09_f09.001       atm  cam.h1   
3           True       CTRL  f.e11.F1850C5CN.f09_f09.001       atm  cam.h1   
4           True       CTRL  f.e11.F1850C5CN.f09_f09.001       atm  cam.h1   

  variable         date_range  member_id  \
0     T200  16000101-16991231          0   
1     T200  24000101-24991231          0   
2     T200  11000101-11991231          0   
3     T200  22000101-22991231          0   
4     T200  25000101-26001231          0   

                                       file_fullpath  \
0  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
1  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
2  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
3  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
4  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   

                                       file_basename  \
0  f.e11.F1850C5CN.f09_f09.001.cam.h1.T200.160001...   
1  f.e11.F1850C5CN.f09_f09.001.cam.h1.T200.240001...   
2  f.e11.F1850C5CN.f09_f09.001.cam.h1.T200.110001...   
3  f.e11.F1850C5CN.f09_f09.001.cam.h1.T200.220001...   
4  f.e11.F1850C5CN.f09_f09.001.cam.h1.T200.250001...   

                                        file_dirname  ctrl_branch_year  \
0  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
1  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
2  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
3  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
4  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   

   year_offset  sequence_order  has_ocean_bgc grid  
0          NaN               0          False  NaN  
1          NaN               0          False  NaN  
2          NaN               0          False  NaN  
3          NaN               0          False  NaN  
4          NaN               0          False  NaN

In [4]:
def collection_summary(col):
    df = col.df
    print("There are:\n")
    print(f"\t1) {df.experiment.nunique()} experiments: \n\n\t\t{df.experiment.unique()}\n")
    print(f"\t2) {df.case.nunique()} cases: \n\n\t\t{df.case.unique()[:5]} etc...\n")
    print(f"\t3) {df.component.nunique()} components: \n\n\t\t{df.component.unique()}\n")
    print(f"\t4) {df.stream.nunique()} streams: \n\n\t\t{df.stream.unique()}\n")
    print(f"\t5) {df.variable.nunique()} variables: \n\n\t\t{df.variable.unique()[:10]} etc...\n")
    
collection_summary(col)

There are:

	1) 3 experiments: 

		['CTRL' '20C' 'RCP85']

	2) 93 cases: 

		['f.e11.F1850C5CN.f09_f09.001' 'e.e11.E1850C5CN.f09_g16.001'
 'b.e11.B1850C5CN.f09_g16.005' 'b.e11.B20TRC5CNBDRD.f09_g16.OIC.010'
 'b.e11.B20TRC5CNBDRD.f09_g16.OIC.009'] etc...

	3) 5 components: 

		['atm' 'lnd' 'rof' 'ice' 'ocn']

	4) 10 streams: 

		['cam.h1' 'cam.h0' 'clm2.h1' 'clm2.h0' 'rtm.h1' 'rtm.h0' 'cice.h1'
 'cice.h' 'pop.h.nday1' 'pop.h']

	5) 985 variables: 

		['T200' 'Q200' 'FLNSC' 'Q500' 'PRECL' 'QBOT' 'TREFHTMN' 'FSNTOA' 'TAUX'
 'TAUY'] etc...



In [5]:
def print_info(cat):
    query_results = cat.query_results
    print(f"This query returned:\n\n{query_results.variable.nunique()} unique variables: {query_results.variable.unique()}\n\n")
    print(f"Number of files: {query_results.file_fullpath.nunique()}\n\n")
    print(f"Date ranges: {query_results.date_range.unique()}\n\n")

## Variables to include in AWS

### Atmosphere, monthly, 3D

In [6]:
atmos_mon_3d = ["T", "U", "V", "Q", "Z3"]

In [7]:
cat = col.search(variable=atmos_mon_3d, component="atm", stream="cam.h0")
print_info(cat)

This query returned:

5 unique variables: ['Q' 'T' 'U' 'V' 'Z3']


Number of files: 880


Date ranges: ['040001-049912' '050001-059912' '060001-069912' '070001-079912'
 '080001-089912' '090001-099912' '100001-109912' '110001-119912'
 '120001-129912' '130001-139912' '140001-149912' '150001-159912'
 '160001-169912' '170001-179912' '180001-189912' '190001-199912'
 '200001-209912' '210001-220012' '185001-200512' '192001-200512'
 '185001-199912' '192001-199912' '010101-019912' '020001-029912'
 '030001-039912' '090001-100112' '000101-009912' '010001-019912'
 '210001-219912' '220001-229912' '230001-239912' '240001-249912'
 '250001-260012' '200601-208012' '208101-210012' '200601-210012']




**NOTE:**
    
- There are **880** files. The spreadsheet mentions **1010** files. 

### Atmosphere, Monthly, Surface

In [8]:
atmos_mon_surface = ["FLNS", "FLNSC", "FLUT", "FSNS", "FSNSC", "FSNTOA", "ICEFRAC", 
                     "LHFLX", "PRECC", "PRECL", "PRECSC", "PRECSL", "PSL", "SHFLX", 
                     "TMQ", "TREFHT", "TREFHTMN", "TREFHTMX", "TS"]

In [9]:
cat = col.search(variable=atmos_mon_surface, component="atm", stream="cam.h0")
print_info(cat)

This query returned:

19 unique variables: ['FLNS' 'FLNSC' 'FLUT' 'FSNS' 'FSNSC' 'FSNTOA' 'ICEFRAC' 'LHFLX' 'PRECC'
 'PRECL' 'PRECSC' 'PRECSL' 'PSL' 'SHFLX' 'TMQ' 'TREFHT' 'TREFHTMN'
 'TREFHTMX' 'TS']


Number of files: 3345


Date ranges: ['040001-049912' '050001-059912' '060001-069912' '070001-079912'
 '080001-089912' '090001-099912' '100001-109912' '110001-119912'
 '120001-129912' '130001-139912' '140001-149912' '150001-159912'
 '160001-169912' '170001-179912' '180001-189912' '190001-199912'
 '200001-209912' '210001-220012' '185001-200512' '192001-200512'
 '185001-199912' '192001-199912' '010101-019912' '020001-029912'
 '030001-039912' '090001-100112' '000101-009912' '010001-019912'
 '210001-219912' '220001-229912' '230001-239912' '240001-249912'
 '250001-260012' '200601-208012' '208101-210012' '200601-210012']




**NOTE:**
    
- There are **3345** files. The spreadsheet mentions **3839** files. 

### Land, Monthly

In [10]:
land_vars = ["FSNO", "H2OSNO", "QRUNOFF", "RAIN", "SNOW", "SOILLIQ", "SOILWATER_10CM"]

In [11]:
cat = col.search(variable=land_vars, component="lnd", stream="clm2.h0")
print_info(cat)

This query returned:

7 unique variables: ['FSNO' 'H2OSNO' 'QRUNOFF' 'RAIN' 'SNOW' 'SOILLIQ' 'SOILWATER_10CM']


Number of files: 1232


Date ranges: ['040001-049912' '050001-059912' '060001-069912' '070001-079912'
 '080001-089912' '090001-099912' '100001-109912' '110001-119912'
 '120001-129912' '130001-139912' '140001-149912' '150001-159912'
 '160001-169912' '170001-179912' '180001-189912' '190001-199912'
 '200001-209912' '210001-220012' '185001-200512' '192001-200512'
 '185001-199912' '192001-199912' '010101-019912' '020001-029912'
 '030001-039912' '090001-100112' '000101-009912' '010001-019912'
 '210001-219912' '220001-229912' '230001-239912' '240001-249912'
 '250001-260012' '200601-208012' '208101-210012' '200601-210012']




### Land, Daily

In [12]:
cat = col.search(variable=land_vars, component="lnd", stream="clm2.h1")
print_info(cat)

This query returned:

6 unique variables: ['FSNO' 'H2OSNO' 'QRUNOFF' 'RAIN' 'SNOW' 'SOILWATER_10CM']


Number of files: 1056


Date ranges: ['04030101-04991231' '05000101-05991231' '06000101-06991231'
 '07000101-07991231' '08000101-08991231' '09000101-09991231'
 '10000101-10991231' '11000101-11991231' '12000101-12991231'
 '13000101-13991231' '14000101-14991231' '15000101-15991231'
 '16000101-16991231' '17000101-17991231' '18000101-18991231'
 '19000101-19991231' '20000101-20991231' '21000101-22001231'
 '18500101-20051231' '19200101-20051231' '18500101-19991231'
 '19200101-19991231' '01010101-01991231' '02000101-02991231'
 '03000101-03991231' '04000101-04991231' '09000101-10011231'
 '00010101-00991231' '01000101-01991231' '21000101-21991231'
 '22000101-22991231' '23000101-23991231' '24000101-24991231'
 '25000101-26001231' '20060101-20801231' '20810101-21001231'
 '20060101-21001231']




**NOTE:**
    
- There are **1056+1232=2288** files. The spreadsheet mentions **1414** files. 

### Ocean, Monthly, Surface

In [13]:
ocn_surface_vars = ["SST", "SSH"]

In [14]:
cat = col.search(variable=ocn_surface_vars, component="ocn", stream="pop.h")
print_info(cat)

This query returned:

2 unique variables: ['SSH' 'SST']


Number of files: 282


Date ranges: ['040001-049912' '050001-059912' '060001-069912' '070001-079912'
 '080001-089912' '090001-099912' '100001-109912' '110001-119912'
 '120001-129912' '130001-139912' '140001-149912' '150001-159912'
 '160001-169912' '170001-179912' '180001-189912' '190001-199912'
 '200001-209912' '210001-220012' '185001-200512' '192001-200512'
 '185001-199912' '192001-199912' '200601-208012' '208101-210012'
 '200601-210012']




### Ocean, Monthly, 3D

In [15]:
ocn_3d_vars = ["SALT"]
cat = col.search(variable=ocn_3d_vars, component="ocn", stream="pop.h")
print_info(cat)

This query returned:

1 unique variables: ['SALT']


Number of files: 141


Date ranges: ['040001-049912' '050001-059912' '060001-069912' '070001-079912'
 '080001-089912' '090001-099912' '100001-109912' '110001-119912'
 '120001-129912' '130001-139912' '140001-149912' '150001-159912'
 '160001-169912' '170001-179912' '180001-189912' '190001-199912'
 '200001-209912' '210001-220012' '185001-200512' '192001-200512'
 '185001-199912' '192001-199912' '200601-208012' '208101-210012'
 '200601-210012']




**NOTE:**
    
- There are **282+141=423** files. The spreadsheet mentions **438** files. 

### Ice, Monthly

In [16]:
ice_mon_vars = ["aice_nh", "aice_sh", "hi_nh", "hi_sh"]
cat = col.search(variable=ice_mon_vars, component="ice", stream="cice.h")
print_info(cat)

This query returned:

4 unique variables: ['aice_nh' 'aice_sh' 'hi_nh' 'hi_sh']


Number of files: 600


Date ranges: ['040001-049912' '050001-059912' '060001-069912' '070001-079912'
 '080001-089912' '090001-099912' '100001-109912' '110001-119912'
 '120001-129912' '130001-139912' '140001-149912' '150001-159912'
 '160001-169912' '170001-179912' '180001-189912' '190001-199912'
 '200001-209912' '210001-220012' '185001-200512' '192001-200512'
 '185001-199912' '192001-199912' '010101-019912' '020001-029912'
 '030001-039912' '090001-100112' '200601-208012' '208101-210012'
 '200601-210012']




**NOTE:**
    
- There are **600** files. The spreadsheet mentions **620** files. 